plugins/modules/kutils.py

```diff
@@ -0,0 +1,372 @@
+      - V(present) is equivalent of C(kinit) command.
+      - V(absent) is equivalent of C(kdestroy) command.
+    type: str
+    default: present
+    choices: ["present", "absent"]
+  kdestroy_all:
+    description:
+      - When O(state=absent) destroys all credential caches in collection.
+      - Equivalent of running C(kdestroy -A).
+    type: bool
+  cache_name:
+    description:
+      - Use O(cache_name) as the ticket cache name and location.
+      - If this option is not used, the default cache name and location are used.
+      - The default credentials cache may vary between systems.
+      - If not set the the value of E(KRB5CCNAME) environment variable will be used instead, its value is used to name the default ticket cache.
+    type: str
+  lifetime:
+    description:
+      - Requests a ticket with the lifetime, if the O(lifetime) is not specified, the default ticket lifetime is used.
+      - Specifying a ticket lifetime longer than the maximum ticket lifetime (configured by each site) will not override the configured maximum ticket lifetime.
+      - "The value for O(lifetime) must be followed by one of the following delimiters: V(s) - seconds, V(m) - minutes, V(h) - hours, V(d) - days."
+      - You cannot mix units; a value of V(3h30m) will result in an error.
+    type: str
+  start_time:
+    description:
+      - Requests a postdated ticket.
+      - Postdated tickets are issued with the invalid flag set, and need to be resubmitted to the KDC for validation before use.
+      - O(start_time) specifies the duration of the delay before the ticket can become valid.
+      - You can use absolute time formats, for July 2, 2024, 1:35:30 p.m. you can use V(240702133530) (yymmddhhmm[ss]).
+      - You can also use time duration format similar to O(lifetime) or O(renewable).
+    type: str
+  renewable:
+    description:
+      - Requests renewable tickets, with a total lifetime equal to O(renewable).
+      - "The value for O(renewable) must be followed by one of the following delimiters: V(s) - seconds, V(m) - minutes, V(h) - hours, V(d) - days."
+      - You cannot mix units; a value of V(3h30m) will result in an error.
+    type: str
+  forwardable:
+    description:
+      - Request forwardable or non-forwardable tickets.
+    type: bool
+  proxiable:
+    description:
+      - Request proxiable or non-proxiable tickets.
+    type: bool
+  address_restricted:
+    description:
+      - Request tickets restricted to the host's local address or non-restricted.
+    type: bool
+  anonymous:
+    description:
+      - Requests anonymous processing.
+    type: bool
+  canonicalization:
+    description:
+      - Requests canonicalization of the principal name, and allows the KDC to reply with a different client principal from the one requested.
+    type: bool
+  enterprise:
+    description:
+      - Treats the principal name as an enterprise name (implies the O(canonicalization) option).
```

Comment:
This could/shoud be enforced in the code - apparently it is not.

Reply:
Honestly I never used this parameter for a `kinit`, but if I got right then the `kinit` command will do it itself and no need to manually imply `-C` if `-E` is set 
![image](https://github.com/user-attachments/assets/6fe9b890-7b49-4446-bda3-2015f6ac8cdd)

Reply:
fair enough :-)